### DISCUSSION TITLE
compiling fprime-system-reference for non-linux embedded targets

### DISCUSSION THREAD

**Discussion by capsulecorplab** [fprime-community/fprime-system-reference#43] [General]
Is there an approach for how one might go about compiling the fprime-system-reference for non-linux embedded target, s.a., a Teensy, Arduino, or ESP device? I realize there's the [fprime-arduino](https://github.com/fprime-community/fprime-arduino) project, but I'm not sure if it's still being maintained

**Comment by Joshua-Anderson** [comment]
A teensy/Arduino/ESP demo of the system reference would be super cool! If the driver implementation and OS abstraction layer already existed for the target, all that would be needed is a separate deployment for that target. I'm not aware of any complete open source bare metal fprime implementations at the moment, but once that was developed adding system reference support wouldn't be too much work. As you mentioned, there is an Arduino implementation, but I think it needs to be updated to the latest version of fprime.

**Reply by capsulecorplab** [reply]
I've tried building the fprime-arduino implementation on v3.1.1 on https://github.com/capsulecorplab/fprime-arduino/tree/test-arduinoblink-on-v3.1.1 and published an immutable virtual desktop environment with the build tools on https://github.com/capsulecorplab/kasm-fprime-workspace/tree/install-cross-compile-toolchain-for-teensy
I kept running into a prescan error (which I should probably go open an issue on...)

**Reply by capsulecorplab** [reply]
Opened issue https://github.com/fprime-community/fprime-arduino/issues/8
